packages/amplify-e2e-tests/src/__tests__/init-special-case.test.ts

```diff
@@ -72,4 +73,11 @@ describe('amplify init', () => {
       getParameters(projectRoot, category, authResourceName);
     }).not.toThrow();
   });
+
+  it('should fail if running amplify init -y on already initialized project', async () => {
+    const envName = 'devtest';
+    const resourceName = 'twoInitDefaultTest';
+    await initJSProjectWithProfile(projectRoot, { disableAmplifyAppCreation: false, name: resourceName, envName });
+    await expect(initNonInteractive(projectRoot)).rejects.toThrowError('Process exited with non zero exit code 1');
```

Comment:
Can you not just run `initJSProjectWithProfile` twice and expect it to throw?

Reply:
`initJSProjectWithProfile` is interactive.

Reply:
The `Invalid environment name: undefined` error only occurs for non-interactive init with `amplify init -y` where envName isn't passed in as an option or through the input prompt

Reply:
pretty weird edge case. But thanks for the clarification